src/net.cpp

```diff
@@ -1221,18 +1221,17 @@ void CConnman::NotifyNumConnectionsChanged()
     }
 }
 
+bool CConnman::RunInactivityChecks(const CNode& node) const
+{
+    return GetSystemTimeInSeconds() > node.nTimeConnected + m_peer_connect_timeout;
```

Comment:
Could make this

```c++
bool CConnman::RunInactivityChecks(const CNode& node, const uint64_t const* pnow=nullptr)
{
    const uint64_t now = (pnow ? *pnow : GetSystemTimeInSeconds());
    return now > node.nTimeconnected + m_peer_connect_timeout;
}
```

and call it from within `InactivityCheck()` as `if (!RunInactivityChecks(node, &now) return false;` to avoid unnecessarily getting the system time twice in quick succession. Might also make `RunInactivityChecks` easier to test since you could pass in different times and check the answer, without having to worry that `GetSystemTime` isn't mockable.

Reply:
#21198